terraform-ecs/roles.tf

```diff
@@ -0,0 +1,122 @@
+# Execution role, permissions to log to cloudwatch
+resource "aws_iam_role" "loki_execution_role" {
+  name = "loki-execution-role"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        }
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy" "loki_execution_policy" {
+  name        = "loki-execution-policy"
+  description = "Permissions for ECS task execution, including logging to CloudWatch"
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ]
+        Resource = "*"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "loki_execution_role_attachment" {
+  role       = aws_iam_role.loki_execution_role.name
+  policy_arn = aws_iam_policy.loki_execution_policy.arn
+}
+
+# Task role for running container, needs to connect to the s3 bucket for logs
+resource "aws_iam_role" "loki_task_role" {
+  name = "loki-task-role"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        }
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy" "loki_task_s3_policy" {
+  name        = "loki-task-policy-s3"
+  description = "Permissions for Loki to access S3"
+  policy = jsonencode({
+    Version = "2012-10-17"
```

Comment:
Do we want to store the version as a variable?

Reply:
We could, this doesn't really change often though, they had a 2008 and then a 2012 and it hasn't changed since: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_version.html